drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -313,97 +313,43 @@ static inline int stm32_clock_control_on(const struct device *dev,
 					 clock_control_subsys_t sub_system)
 {
 	struct stm32_pclken *pclken = (struct stm32_pclken *)(sub_system);
-	int rc = 0;
+	volatile uint32_t *reg;
+	uint32_t reg_val;
 
 	ARG_UNUSED(dev);
 
-	/* Both cores can access banks by following LL API */
-	/* Using "_Cn_" LL API would restrict access to one or the other */
 	z_stm32_hsem_lock(CFG_HW_RCC_SEMID, HSEM_LOCK_DEFAULT_RETRY);
-	switch (pclken->bus) {
-	case STM32_CLOCK_BUS_AHB1:
-		LL_AHB1_GRP1_EnableClock(pclken->enr);
-		break;
-	case STM32_CLOCK_BUS_AHB2:
-		LL_AHB2_GRP1_EnableClock(pclken->enr);
-		break;
-	case STM32_CLOCK_BUS_AHB3:
-		LL_AHB3_GRP1_EnableClock(pclken->enr);
-		break;
-	case STM32_CLOCK_BUS_AHB4:
-		LL_AHB4_GRP1_EnableClock(pclken->enr);
-		break;
-	case STM32_CLOCK_BUS_APB1:
-		LL_APB1_GRP1_EnableClock(pclken->enr);
-		break;
-	case STM32_CLOCK_BUS_APB1_2:
-		LL_APB1_GRP2_EnableClock(pclken->enr);
-		break;
-	case STM32_CLOCK_BUS_APB2:
-		LL_APB2_GRP1_EnableClock(pclken->enr);
-		break;
-	case STM32_CLOCK_BUS_APB3:
-		LL_APB3_GRP1_EnableClock(pclken->enr);
-		break;
-	case STM32_CLOCK_BUS_APB4:
-		LL_APB4_GRP1_EnableClock(pclken->enr);
-		break;
-	default:
-		rc = -ENOTSUP;
-		break;
-	}
+
+	reg = (uint32_t *)(DT_REG_ADDR(DT_NODELABEL(rcc)) + pclken->bus);
+	reg_val = *reg;
+	reg_val &= ~pclken->enr;
+	reg_val |= pclken->enr;
+	*reg = reg_val;
 
 	z_stm32_hsem_unlock(CFG_HW_RCC_SEMID);
 
-	return rc;
+	return 0;
```

Comment:
is there any benefit to raise an error code here in case the DTS is malformed ?

Reply:
In a next commit, a check is added to verify that the requested clock is enabled.
Regarding the other parameters I'm not sure we can do much more checks.